src/components/CarListComponent/CarListComponent.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import {
+  Order_By,
+  Query_GetCarsQueryVariables,
+  useQuery_GetCarsQuery,
+} from '../../api/graphql/__generated__/graphql-types';
+import { isValidUuid } from '../../helpers/validators';
+import ErrorMessage from '../ErrorMessage/ErrorMessage';
+import FilterComponent from '../FilterComponent/FilterComponent';
+import HeaderList from '../HeaderList/HeaderList';
+import LoadingComponent from '../LoadingComponent/Loading';
+import BodyList from './BodyList/BodyList';
+import { Container } from './CarListComponentStyled';
+
+interface usercars {
+  user_id: number;
+  car_id: number;
+}
+
+interface CarListComponentProps {
+  favoritesCarsArray?: usercars[];
+}
+
+const CarListComponent = ({ favoritesCarsArray }: CarListComponentProps) => {
+  const { data, loading, error, refetch } = useQuery_GetCarsQuery();
+  const [searchParam, setSearchParam] = useSearchParams();
+  const [filterObject, setFilterObject] =
+    useState<Query_GetCarsQueryVariables>();
+  useEffect(() => {
+    if (isValidUuid(searchParam.get('search')!)) {
+      setFilterObject({
+        orderBy: [
+          {
+            sale_date: searchParam.get('order')
+              ? searchParam.get('order') == 'desc'
+                ? Order_By.Desc
+                : Order_By.Asc
+              : Order_By.Desc,
+          },
+        ],
+        where: {
+          _and: [
+            {
+              batch: {
+                _eq: searchParam.get('search')!,
+              },
+            },
+            {
+              id: {
+                _in:
+                  favoritesCarsArray &&
+                  favoritesCarsArray.map((carId) => carId.car_id),
+              },
+            },
+          ],
+        },
+      });
+    } else {
+      setFilterObject({
+        orderBy: [
+          {
+            sale_date: searchParam.get('order')
+              ? searchParam.get('order') == 'desc'
+                ? Order_By.Desc
+                : Order_By.Asc
+              : Order_By.Desc,
+          },
+        ],
+        where: {
+          _or: [
+            {
+              title: {
+                _iregex: searchParam.get('search')
+                  ? searchParam.get('search')
+                  : '',
+              },
+            },
+            {
+              vin: {
+                _iregex: searchParam.get('search')
+                  ? searchParam.get('search')
+                  : '',
+              },
+            },
+          ],
+          id: {
+            _in:
+              favoritesCarsArray &&
+              favoritesCarsArray.map((carId) => carId.car_id),
+          },
+        },
+      });
+    }
+  }, [searchParam, favoritesCarsArray]);

```

Comment:
is `searchParam` something a user types? why don't you handle the request when the input is changing instead of having the url param as dependency on a useEffect? also you have 2 useEffects that can be simplify into one or by using the approach I mentioned

Reply:
Reason: searchParam is a useParams variable when the user click search i set the input value and the order value to the params, when params change i can refetch the query, i did it like that trying to avoid pass to many props to filter components that has te filter logic to send data to params, and related to use effect, could be just 1 my bad, 

Solution: 
I will delete the second useEffect and just refetch the query removing an unnecessary filterObject variable, and regarding filter when user types, i will remove the search button and will implement on text change logic and implement debounce avoiding too many request typing, about searchParam dependency on my useEffect right now i think it is necessary because is my way to know if something change on filter component. but i will find a solution to avoid to put searchparam as dependency

Reply:
If this query is to be run by search params, rather accomplishing it from a refetch, a simpler approach would be to pipe it directly into the query.
(winging it on syntax here 😂 )

```
const [searchParams, setSearchParams] = useParams()

const search = searchParams.get('search')

const orderBy = searchParams.get('search')

const { data, loading, error, refetch } = useQuery_GetCarsQuery( { variables: { search, orderBy } } );

```